### PR TITLE
Replaced iothub service with azure-iot, using MQTT, no dependencies

### PR DESCRIPTION
**Pull request by mortenf**
As per #391, the old "iothub" service is defunct. This change provides a new service, "azure-iot", using the MQTT protocol and thus without dependencies.

### CODE REVIEW

**Comment by amotl** [comment]
Dear Morten,

thanks a bunch for your contribution!

I have just a minor remark: Can I humbly ask you to name the module `azure_iot.py`? While it currently works to use dashes in module names, I am aiming to simplify the plugin loading at some time in the future where this feature might be dropped.

With kind regards,
Andreas.

File: mqttwarn/services/azure-iot.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    # iot hub name is stored in config
+    iothubname = item.config['iothubname']
+
+    # connect...
+    params = {
+        'hostname': iothubname + ".azure-devices.net",
+        'port': 8883,
+        'protocol': mqttclient.MQTTv311,
+        'qos': 0,
```
Comment (amotl) [inline mqttwarn/services/azure-iot.py:28]: Is it possible to use an elevated QoS value here? If so, we might want to make configurable:
```
'qos': int(item.config.get('qos', 0))
```

Comment (mortenf) [reply mqttwarn/services/azure-iot.py]: Sure. QoS 2 is not allowed, however, so I've added a check.

**Comment by amotl** [comment]
This has been merged through #448. Thanks a bunch, @mortenf!

**Pull request closed without merging**